glotaran/project/test/test_project.py

```diff
@@ -0,0 +1,168 @@
+from pathlib import Path
+
+import pytest
+
+from glotaran import __version__ as gta_version
+from glotaran.io import save_dataset
+from glotaran.io import save_parameters
+from glotaran.project.project import Project
+from glotaran.project.result import Result
+from glotaran.testing.simulated_data.sequential_spectral_decay import DATASET as example_dataset
+from glotaran.testing.simulated_data.sequential_spectral_decay import (
+    MODEL_YML as example_model_yml,
+)
+from glotaran.testing.simulated_data.sequential_spectral_decay import (
+    PARAMETERS as example_parameter,
+)
+
+
+@pytest.fixture(scope="module")
+def project_folder(tmpdir_factory):
+    return str(tmpdir_factory.mktemp("test_project"))
+
+
+@pytest.fixture(scope="module")
+def project_file(project_folder):
+    return Path(project_folder) / "project.gta"
+
+
+@pytest.fixture(scope="module")
+def test_data(tmpdir_factory):
+    path = Path(tmpdir_factory.mktemp("test_project")) / "dataset_1.nc"
+    save_dataset(example_dataset, path)
+    return path
+
+
+def test_open(project_folder, project_file):
+    project_from_folder = Project.open(project_folder)
+
+    project_from_file = Project.open(project_file)
+
+    assert project_from_folder == project_from_file
+
+    project = project_from_file
+
+    assert project.version == gta_version
+    assert not project.has_models
+    assert not project.has_data
+    assert not project.has_parameters
+    assert not project.has_results
+
+
+def test_generate_model(project_folder, project_file):
+    project = Project.open(project_file)
+
+    project.generate_model("test_model", "decay_parallel", {"nr_compartments": 5})
+
+    model_folder = Path(project_folder) / "models"
```

Comment:
Redundant `Path` initialization since `project_folder` always is a `Path` (See other copypasta occurrences)